kpi/serializers/current_user.py

```diff
@@ -75,18 +75,47 @@ def get_git_rev(self, obj):
     def to_representation(self, obj):
         if obj.is_anonymous:
             return {'message': 'user is not logged in'}
+
         rep = super().to_representation(obj)
-        if not rep['extra_details']:
+        if not rep['extra_details'] or not isinstance(
+            rep['extra_details'], dict
+        ):
             rep['extra_details'] = {}
-        elif rep.get('primarySector'):
-            # the front end used to set this with camelCase but has since
-            # been fixed
-            rep['primary_sector'] = rep['primarySector']
-            del rep['primarySector']
+        extra_details = rep['extra_details']
+
+        # the front end used to set `primarySector` but has since been changed
+        # to `sector`, which matches the registration form
+        if extra_details.get('primarySector') and not extra_details.get(
+            'sector'
+        ):
```

Comment:
I would have the same suggestion for this `if` statement as said above .

Link To be copied

Reply:
ok, done